File: src/main/java/com/lannstark/lec12/Lec12Main.java
package com.lannstark.lec12;

public class Lec12Main {

  public static void main(String[] args) {
    Person.Factory.newBaby("ABC");
    
    // @JvmStatic을 붙였을 때 이렇게 사용 가능
    Person.newBaby("ABC");
  }

}

File: src/main/java/com/lannstark/lec12/Lec12Main.kt
package com.lannstark.lec12

fun main() {
    moveSomething(object  : Movable {
        override fun move() {
            println("이동합니다.")
        }

        override fun fly() {
            println("날아갑니다.")
        }
    })
}

private fun moveSomething(movable: Movable) {
    movable.move()
    movable.fly()
}